core-bundle/src/Security/Authentication/Token/FrontendPreviewToken.php

```diff
@@ -23,7 +24,7 @@ public function __construct(?FrontendUser $user, bool $showUnpublished)
     {
         if (null === $user) {
             parent::__construct();
-            $this->setUser('anon.');
+            $this->setUser(new InMemoryUser('anon.', null));
```

Comment:
According to Symfony, _anonymous is now represented by the absence of a token_.

https://github.com/symfony/symfony/blob/e0fdec0f739f7324e1338940a5b42c787a81167b/src/Symfony/Component/Security/Core/Authentication/Token/AnonymousToken.php#L21

Shouldn't the code be something like this then?

```php
if (null === $user) {
    parent::__construct();
} else {
    parent::__construct($user->getRoles());
    $this->setUser($user);
    $this->setAuthenticated(true);
}

$this->showUnpublished = $showUnpublished;
```

@aschempp /cc

Reply:
Correct 👍

Reply:
Fixed in d485127b6.

Reply:
Had to undo e33d60c in 530a62c6f. If we change to `anonymous is now represented by the absence of a token`, we have to adjust more or less all security related stuff.